3_summarize/src/annual_thermal_metrics.R

```diff
@@ -0,0 +1,256 @@
+    } else {
+      wtr_bot <- NA
+    }
+    return(wtr_bot)
+  }) %>% 
+    purrr::reduce(c) %>% 
+    max(na.rm = TRUE)
+}
+
+## Need to do the following fxns for each month.
+# TODO: think strategically about how to implement for each month.
+
+#' @description mean of surface temperature for a month
+mean_surf_mon <- function(date, wtr, depth) {
+  data.frame(date, month = tolower(format(date, "%b")), depth = depth, wtr = wtr) %>% 
+    # Get just surface temps
+    filter(depth == 0) %>% 
+    
+    # Sort months in annual order
+    arrange(as.numeric(format(date, "%m"))) %>%
+    
+    # Find mean for each month
+    group_by(month) %>% 
+    summarize(mean_surf = mean(wtr, na.rm = TRUE), .groups = "keep") %>% 
+    pivot_wider(names_from = month, values_from = mean_surf, names_prefix = "mean_surf_")
+}
+
+#' @description max of surface temperature for each month
+max_surf_mon <- function(date, wtr, depth) {
+  data.frame(date, month = tolower(format(date, "%b")), depth = depth, wtr = wtr) %>% 
+    # Get just surface temps
+    filter(depth == 0) %>% 
+    
+    # Sort months in annual order
+    arrange(as.numeric(format(date, "%m"))) %>%
+    
+    # Find max for each month
+    group_by(month) %>% 
+    summarize(max_surf = max(wtr, na.rm = TRUE), .groups = "keep") %>% 
+    pivot_wider(names_from = month, values_from = max_surf, names_prefix = "max_surf_") %>% 
+    ungroup()
+}
+
+#' @description mean of bottom temperature for each month
+mean_bot_mon <- function(date, wtr, depth) {
+  data.frame(date, month = tolower(format(date, "%b")), depth = depth, wtr = wtr) %>% 
+    
+    # Calculate water temp at the bottom
+    group_by(date) %>% 
+    mutate(wtr_bot = find_wtr_at_lake_bottom(wtr, depth)) %>% 
+    ungroup() %>% 
+    
+    # Sort months in annual order
+    arrange(as.numeric(format(date, "%m"))) %>% 
+    
+    # Then find the mean for each month
+    group_by(month) %>% 
+    summarize(mean_bot = mean(wtr_bot, na.rm = TRUE), .groups = "keep") %>% 
+    pivot_wider(names_from = month, values_from = mean_bot, names_prefix = "mean_bot_") %>% 
+    ungroup()
+}
+
+#' @description max of bottom temperature for each month
+max_bot_mon <- function(date, wtr, depth) {
+  data.frame(date, month = tolower(format(date, "%b")), depth = depth, wtr = wtr) %>% 
+    # Calculate water temp at the bottom
+    group_by(date) %>% 
+    mutate(wtr_bot = find_wtr_at_lake_bottom(wtr, depth)) %>% 
+    ungroup() %>%  
+    
+    # Sort months in annual order
+    arrange(as.numeric(format(date, "%m"))) %>% 
+    
+    # Then find the mean for each month
+    group_by(month) %>% 
+    summarize(max_bot = max(wtr_bot, na.rm = TRUE), .groups = "keep") %>% 
+    pivot_wider(names_from = month, values_from = max_bot, names_prefix = "max_bot_") %>% 
+    ungroup()
+}
+
+## Helper functions for the above
+
+calc_gdd <- function(wtr, base = 0) {
+  sum(wtr[wtr > base])
```

Comment:
The description of this metric said cumulative sum ... but I assumed we just wanted one value for the year so that would really just be the sum total.

Reply:
yes